tests/e2e/p/staking_rewards.go

```diff
@@ -0,0 +1,284 @@
+				pWallet.AVAXAssetID(),
+				&secp256k1fx.OutputOwners{
+					Threshold: 1,
+					Addrs:     []ids.ShortID{deltaDelegationRewardKey.Address()},
+				},
+			)
+			require.NoError(err)
+		})
+
+		ginkgo.By("stopping beta node to prevent it and its delegator from receiving a validation reward")
+		require.NoError(betaNode.Stop())
+
+		ginkgo.By("waiting until all validation periods are over")
+		// The beta validator was the last added and so has the latest end time. The
+		// delegation periods are shorter than the validation periods.
+		time.Sleep(time.Until(betaValidatorEndTime))
+
+		pvmClient := platformvm.NewClient(alphaNode.GetProcessContext().URI)
+
+		ginkgo.By("waiting until the alpha and beta nodes are no longer validators")
+		e2e.Eventually(func() bool {
+			validators, err := pvmClient.GetCurrentValidators(e2e.DefaultContext(), ids.Empty, nil)
+			require.NoError(err)
+			for _, validator := range validators {
+				if validator.NodeID == alphaNodeID || validator.NodeID == betaNodeID {
+					return false
+				}
+			}
+			return true
+		}, e2e.DefaultTimeout, e2e.DefaultPollingInterval, "nodes failed to stop validating before timeout ")
+
+		ginkgo.By("retrieving reward configuration for the network")
+		// TODO(marun) Enable GetConfig to return *node.Config
+		// directly. Currently, due to a circular dependency issue, a
+		// map-based equivalent is used for which manual unmarshaling
+		// is required.
+		adminClient := admin.NewClient(e2e.Env.GetRandomNodeURI().URI)
+		rawNodeConfigMap, err := adminClient.GetConfig(e2e.DefaultContext())
+		require.NoError(err)
+		nodeConfigMap, ok := rawNodeConfigMap.(map[string]interface{})
+		require.True(ok)
+		stakingConfigMap, ok := nodeConfigMap["stakingConfig"].(map[string]interface{})
+		require.True(ok)
+		rawRewardConfig := stakingConfigMap["rewardConfig"]
+		rewardConfig := reward.Config{}
+		require.NoError(mapstructure.Decode(rawRewardConfig, &rewardConfig))
+
+		ginkgo.By("retrieving reward address balances")
+		rewardBalances := make(map[ids.ShortID]uint64, len(rewardKeys))
+		for _, rewardKey := range rewardKeys {
+			keychain := secp256k1fx.NewKeychain(rewardKey)
+			baseWallet := e2e.Env.NewWallet(keychain, nodeURI)
+			pWallet := baseWallet.P()
+			balances, err := pWallet.Builder().GetBalance()
+			require.NoError(err)
+			rewardBalances[rewardKey.Address()] = balances[pWallet.AVAXAssetID()]
+		}
+		require.Len(rewardBalances, len(rewardKeys))
+
+		ginkgo.By("determining expected validation and delegation rewards")
+		currentSupply, _, err := pvmClient.GetCurrentSupply(e2e.DefaultContext(), constants.PrimaryNetworkID)
+		require.NoError(err)
+		calculator := reward.NewCalculator(rewardConfig)
+		expectedValidationReward := calculator.Calculate(validationPeriod, weight, currentSupply)
+		expectedDelegationReward := calculator.Calculate(delegationPeriod, weight, currentSupply)
+		expectedDelegationFee := uint64(math.Round(float64(expectedDelegationReward) * delegationPercent))
```

Comment:
on dev we just introduced the reward.Split utility to handle splitting the shares. I'd suggest you use that to avoid issues with rounding (which it treated differently).